metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -173,6 +175,127 @@ private void testDelayedConfigurationOperations(LocalLogManagerTestEnv logEnv,
+                    createTopicsResponseData.topics().find("foo").errorCode());
+                assertEquals("Unable to replicate the partition " + replicationFactor + " time(s): All brokers " +
+                    "are currently fenced.", createTopicsResponseData.topics().find("foo").errorMessage());
+
+                // Unfence all brokers
+                sendBrokerheartbeat(active, brokerCount, brokerEpochs);
+                createTopicsResponseData = active.createTopics(
+                    createTopicsRequestData).get();
+                assertEquals(Errors.NONE.code(), createTopicsResponseData.topics().find("foo").errorCode());
+                Uuid topicIdFoo = createTopicsResponseData.topics().find("foo").topicId();
+                sendBrokerheartbeat(active, brokerCount, brokerEpochs);
+                Thread.sleep(sleepMillis);
+
+                // All brokers should still be unfenced
+                for (int brokerId = 0; brokerId < brokerCount; brokerId++) {
+                    assertTrue(active.replicationControl().isBrokerUnfenced(brokerId),
+                        "Broker " + brokerId + " should have been unfenced");
+                }
+                createTopicsRequestData = new CreateTopicsRequestData().setTopics(
+                        new CreatableTopicCollection(Collections.singleton(
+                            new CreatableTopic().setName("bar").setNumPartitions(1).
+                                setConfigs(new CreateableTopicConfigCollection(Collections.
+                                    singleton(new CreateableTopicConfig().setName("min.insync.replicas").
+                                        setValue("2")).iterator())).
+                                setReplicationFactor(replicationFactor)).iterator()));
+                createTopicsResponseData = active.createTopics(createTopicsRequestData).get();
+                assertEquals(Errors.NONE.code(), createTopicsResponseData.topics().find("bar").errorCode());
+                Uuid topicIdBar = createTopicsResponseData.topics().find("bar").topicId();
+
+                // Fence some of the brokers
+                boolean fencingComplete;
+                Long waitIterations = 0L;
+                do {
+                    fencingComplete = true;
+                    sendBrokerheartbeat(active, brokersToKeepUnfenced, brokerEpochs);
+                    for (int i = brokersToKeepUnfenced; i < brokerCount; i++) {
+                        if (active.replicationControl().isBrokerUnfenced(i)) {
+                            fencingComplete = false;
+                        }
+                    }
+                    Thread.sleep(1000);
+                    waitIterations++;
+
+                    if (waitIterations >= sessionTimeout * 3) {
+                        assertTrue(false, "Fencing of brokers did not process within expected time");
+                    }
+                } while (!fencingComplete);
```

Comment:
Maybe you can replace this `while` loop with a util function `TestUtils.waitForCondition`. Ex:
`TestUtils.waitForCondition(() -> fenceBrokers(), 3000, "Fencing of brokers did not process within expected time");`
Just that it only accepted the `timeout` value, not the `waitIterations` like you did. FYI

Reply:
Will check this out. Thanks for the pointer.